WooCommerce/WooCommerceTests/ViewRelated/Shipping Label/Create Shipping Label/ShippingLabelFormViewModelTests.swift

```diff
@@ -71,9 +71,72 @@ final class ShippingLabelFormViewModelTests: XCTestCase {
                                                            postcode: "94121-2303")
 
         // When
-        shippingLabelFormViewModel.handleDestinationAddressValueChanges(address: expectedShippingAddress)
+        shippingLabelFormViewModel.handleDestinationAddressValueChanges(address: expectedShippingAddress, validated: true)
 
         // Then
         XCTAssertEqual(shippingLabelFormViewModel.destinationAddress, expectedShippingAddress)
     }
+
+    func test_sections_returns_updated_rows_after_validating_origin_address() {
+        // Given
+        let shippingLabelFormViewModel = ShippingLabelFormViewModel(siteID: 10, originAddress: nil, destinationAddress: nil)
+        let expectedShippingAddress = MockShippingLabelAddress.sampleAddress()
+        let currentRows = shippingLabelFormViewModel.state.sections.first?.rows
+
+        // When
+        XCTAssertEqual(currentRows?[0].type, .shipFrom)
+        XCTAssertEqual(currentRows?[0].dataState, .pending)
+        XCTAssertEqual(currentRows?[0].displayMode, .editable)
+        XCTAssertEqual(currentRows?[1].type, .shipTo)
+        XCTAssertEqual(currentRows?[1].dataState, .pending)
+        XCTAssertEqual(currentRows?[1].displayMode, .disabled)
+        XCTAssertEqual(currentRows?[2].type, .packageDetails)
+        XCTAssertEqual(currentRows?[2].dataState, .pending)
+        XCTAssertEqual(currentRows?[2].displayMode, .disabled)
```

Comment:
nit: maybe asserts for initial state should belong to `Given` section, so `When` is more focused on action being tested?

Reply:
yes, it sounds right 👍 Updated here ab83f198ac68d0eae0799b09a89e8cbd8b047203